crates/worker/src/config.rs

```diff
@@ -488,3 +530,59 @@ impl ValidatableConfig for Config {
         Ok(())
     }
 }
+
+/// Converts a semantic version string to a Python PEP 440-ish version string
+/// aligning it with the respective CI action.
+fn to_pyver(input: &str) -> String {
+    if let Some((base, rest)) = input.split_once("-alpha.") {
+        return format!("{base}a{rest}");
+    }
+
+    if let Some((base, rest)) = input.split_once("-beta.") {
+        return format!("{base}b{rest}");
+    }
+
+    if let Some((base, rest)) = input.split_once("-rc.") {
+        return format!("{base}rc{rest}");
+    }
+
+    match input.split_once('-') {
+        Some((base, _suffix)) => base.to_string(),
+        None => input.to_string(),
+    }
+}
+
+#[cfg(test)]
```

Comment:
since there are tests, its fair to assume this is more than just a yolo quickfix. now we should make sure the entire semver standard is covered before this is merged. /s